kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/SchemaRegistryService.java

```diff
@@ -372,12 +379,62 @@ private WebClient.RequestBodySpec configuredWebClient(KafkaCluster cluster,
                                                         List<String> uriVariables,
                                                         MultiValueMap<String, String> queryParams) {
     final var schemaRegistry = cluster.getSchemaRegistry();
-    return webClient
+
+    return securedWebClientOnTls(cluster)
         .method(method)
         .uri(buildUri(schemaRegistry, path, uriVariables, queryParams))
         .headers(headers -> setBasicAuthIfEnabled(schemaRegistry, headers));
   }
 
+  private WebClient securedWebClientOnTls(KafkaCluster cluster) {
+    InternalSchemaRegistry srConfig = cluster.getSchemaRegistry();
+
+    // If we want to customize our TLS configuration, we need at least a truststore
+    if (srConfig.getTrustStoreLocation() == null || srConfig.getTrustStorePassword() == null) {
+      return WebClient.builder().build();
+    }
+
+    try {
+      SslContextBuilder contextBuilder = SslContextBuilder.forClient();
+
+      // Prepare truststore
+      KeyStore trustStore = KeyStore.getInstance("JKS");
+      trustStore.load(
+              new FileInputStream((ResourceUtils.getFile(srConfig.getTrustStoreLocation()))),
+              srConfig.getTrustStorePassword().toCharArray()
+      );
+
+      TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(
+          TrustManagerFactory.getDefaultAlgorithm()
+      );
+      trustManagerFactory.init(trustStore);
+      contextBuilder.trustManager(trustManagerFactory);
+
+      // Prepare keystore only if we got a keystore
+      if (srConfig.getKeyStoreLocation() != null && srConfig.getKeyStorePassword() != null) {
+        KeyStore keyStore = KeyStore.getInstance("JKS");
+        keyStore.load(
+                new FileInputStream(ResourceUtils.getFile(srConfig.getKeyStoreLocation())),
+                srConfig.getKeyStorePassword().toCharArray()
+        );
+
+        KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+        keyManagerFactory.init(keyStore, srConfig.getKeyStorePassword().toCharArray());
+        contextBuilder.keyManager(keyManagerFactory);
+      }
+
+      // Create webclient
+      SslContext context = contextBuilder.build();
+
+      return WebClient.builder()
+              .clientConnector(new ReactorClientHttpConnector(HttpClient.create().secure(t -> t.sslContext(context))))
+              .build();
```

Comment:
I think this logic worth to be extracted to further reuse - Lets create new class `SecuredWebClient` with static `configure(...)` method.

Reply:
I extracted it as `util.SecuredWebClient` with the proposed static `configure` method. A notable change is that I now return a builder instead of the built `WebClient` as for example in #2833 it can be more handy to use a builder. I hope the refactoring is done the way you like